src/ic3/m1ain.cc

```diff
@@ -87,6 +113,30 @@ int ic3_enginet::operator()()
       message.error() << "no properties" << messaget::eom;
       return 1;
     }
+
+    std::size_t number_of_properties = 0;
+
+    for(auto &property : properties.properties)
+    {
+      if(property.is_disabled())
+        continue;
+
+      // Is it supported by the IC3 engine?
+      if(!ic3_supports_property(property.normalized_expr))
+      {
+        property.failure("property not supported by IC3 engine");
+        continue;
+      }
+
+      number_of_properties++;
+    }
+
+    if(number_of_properties == 0)
+    {
+      namespacet ns(transition_system.symbol_table);
+      report_results(cmdline, properties, ns, message.get_message_handler());
+      return 10;
+    }
```

Comment:
Can you please explain why "10" is the appropriate return value when there were no supported properties? What if there were no properties at all in the first place?

Reply:
Will extract into a method in a separate PR, which will hopefully give us consistency.